src/util/ParserUtil.h

```diff
@@ -125,6 +125,63 @@ class ParserUtil final {
         }
         pred->setFilter(newFilter);
     }
+
+
+    static void rewriteReduce(QueryContext *qctx,
```

Comment:
What's this purpose?

Reply:
Rewrite it so we can eval it properly. The logic here is as same as  list comprehension and predicate expression